eZ/Publish/Core/Persistence/Legacy/Content/Handler.php

```diff
@@ -587,20 +598,29 @@ public function listVersions($contentId, $status = null, $limit = -1)
      */
     public function copy($contentId, $versionNo = null)
     {
+        $contentInfo = $this->loadContentInfo($contentId);
+
         $currentVersionNo = isset($versionNo) ?
             $versionNo :
-            $this->loadContentInfo($contentId)->currentVersionNo;
+            $contentInfo->currentVersionNo;
 
         // Copy content in given version or current version
         $createStruct = $this->mapper->createCreateStructFromContent(
             $this->load($contentId, $currentVersionNo)
         );
+
+        if (!isset($versionNo) && $currentVersionNo > 1) {
+            $createStruct->initialLanguageId = $this
+                ->languageHandler
+                ->loadByLanguageCode($contentInfo->mainLanguageCode)
+                ->id;
+        }
```

Comment:
It seems to me setting proper `initialLanguageId` is a responsibility of `eZ\Publish\Core\Persistence\Legacy\Content\Mapper::createCreateStructFromContent` and rather should be fixed there, so `$createStruct` returned above this block is correct.